### DISCUSSION TITLE
macos zip utility can't open the zips

### DISCUSSION THREAD

**Discussion by asturur** [gildas-lormeau/zip.js#275] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/1194048/134020015-7db4b800-5bad-431d-918f-c19a6ef78abd.png)

Same zip file can be opened with another zip utility ( i tried keka for example ).

I tried to change the compression level or the zip64 format, but nothing change.
Is this a known issue?

**Comment by gildas-lormeau** [comment, marked as answer]
This software does not support data descriptor signatures. Set the option `dataDescriptorSignature` to `false`. Alternatively you can disable data descriptors by setting the option `dataDescriptor` to `false`.

**Comment by asturur** [comment]
sure let me try right now

**Comment by asturur** [comment]
seems to work. Does this diserve a readme or a changed default value?

**Comment by gildas-lormeau** [comment]
I prefer to let `dataDescriptor` to `true` because it allows to stream the zip file. Maybe moving this thread to the "Discussions" tab could suffice?

**Comment by asturur** [comment]
Yeas please make it a discussion. What does it mean that allow to stream the file?

**Reply by gildas-lormeau** [reply]
By default, if you don't add files in parallel in the zip file then `zip.js` will be able to create the zip file by using a small buffer and appending data without eating all your RAM. This is what I mean when I use the term "streaming".

**Reply by asturur** [reply]
i m queuing a file at time 500 megs up to 3.6gb, and i m doing it one at time and keep buffered false because i don't want to eat the ram.
Is `dataDescriptor: false` blocking me from doing that?

**Reply by gildas-lormeau** [reply]
Yes, because without the data descriptor, the compressed size of the entry must be written *before* the compressed data in the zip file. I would recommend you to keep `dataDescriptor` to `true`. This issue a limitation of the macos unzip utility. Every decent tool implement this feature for at least 10 years.

**Reply by gildas-lormeau** [reply]
See https://pkware.cachefly.net/webdocs/APPNOTE/APPNOTE-6.3.9.TXT#:~:text=4.3.9%20%20Data%20descriptor (part 4.3.9) for more information about the data descriptor in zip files.

**Reply by asturur** [reply]
99% of the zip i download are compatible tho. Is everyone holding back because of macos?
This is a real question i m not trying to counterpoint what you just said

**Reply by gildas-lormeau** [reply]
No, I think that most of the time, RAM is not an issue so most of the zip tools don't rely on data descriptors to optimize it.

**Comment by gildas-lormeau** [comment]
@asturur Can you tell me if you can open the attached file with the Archive Utility?

[no-data-descriptor-signature.zip](https://github.com/gildas-lormeau/zip.js/files/7199158/no-data-descriptor-signature.zip)

**Reply by asturur** [reply]
yes this works

**Reply by gildas-lormeau** [reply]
Interesting! It means that Archive Utility supports data descriptors but without the signature. 

So, you have to update zip.js to the version `2.3.17` and set the option `dataDescriptorSignature` to `false` (you can omit `dataDescriptor` which is `true` by default). I did not set `dataDescriptorSignature` to `false` by default because it would be a breaking change.
